### PR TITLE
State name changes for filter options || Simplified Report Closure

### PR DESCRIPTION
**Pull request by arjunaj5**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=CgRfspq)

Loom recording staging: https://www.loom.com/share/58b98f0ff9804afdb46c6a7a64c09ea8

### CODE REVIEW

File: src/app/fyle/team-reports/team-reports.page.ts [hunk 1]
```diff
@@ -923,6 +941,7 @@ export class TeamReportsPage implements OnInit {
             ],
           } as FilterOptions<string>,
         ],
+        simplifyReportsEnabled$: this.simplifyReportsEnabled$,
```
Comment (yksolanki9) [inline src/app/fyle/team-reports/team-reports.page.ts:944]: same as above

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Passing obvservables to dialog is not a good practive, resolve it in the parent component itself, and pass the result to the dialog

File: src/app/fyle/my-reports/my-reports.page.ts [hunk 2]
```diff
@@ -1010,6 +1058,8 @@ export class MyReportsPage implements OnInit {
             ],
           } as FilterOptions<string>,
         ],
+        simplifyReportsEnabled$: this.simplifyReportsEnabled$,
```
[the remark this answers is not in the document]
Comment (arjunaj5) [reply src/app/fyle/my-reports/my-reports.page.ts:1061]: If we resolve in the parent component itself, we would not be able to show the skeleton loaders as the modal does not react to changes in the `componentProps`. We will have to disable the filter funnel icon in this case. Is this fine?  @yksolanki9

File: src/app/shared/components/fy-filters/fy-filters.component.ts [hunk 3]
```diff
@@ -60,6 +65,18 @@ export class FyFiltersComponent implements OnInit {
       this.startDate = this.customDateMap[this.activeFilter.name]?.startDate;
       this.endDate = this.customDateMap[this.activeFilter.name]?.endDate;
     }
+
+    const stateFilterIndex = this.filterOptions.findIndex((option) => option.name === 'State');
```
Comment (yksolanki9) [inline src/app/shared/components/fy-filters/fy-filters.component.ts:69]: Move this inside forkJoin since it is not used anywhere else

**Review by yksolanki9 — APPROVED** [review]
Minor comment

**Pull request merged**